functions/AutoRemediateS3-002.js

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+const config = require('./config');
+const AWS = require("aws-sdk");
+
+const CCRuleCode = 'S3-002'
+const CCRuleName = 'BucketPublicReadAcpAccess'
+const allUsersURI = 'http://acs.amazonaws.com/groups/global/AllUsers'
+const readAcpPermission = "READ_ACP"
+
+module.exports = {
+
+  removeAcpPermission: function (thisGrant, newAcl) {
+    if (thisGrant.Permission != readAcpPermission) {  // any besides READ_ACP are passed through
+      newAcl['Grants'].push(thisGrant);
+    }
+
+    return newAcl;
+  },
+
+  transferOwner: function (oldAcl, newAcl) {
+    newAcl.Owner = oldAcl.Owner; // transfer the existing bucket owner
+
+    return newAcl;
+  },
+
+  transferAcl: function (oldAcl, newAcl) {
```

Comment:
A better name for this would be `filterAcl` Since it filters violating grants.

`aclNew` does not need to be passed in.  This function should return a new ACL, based on the ACL passed in, but with violating permissions filtered out.

Reply:
`filterAcl` could then be made more general purpose by passing in the grant to remove:

```
// Returns a copy of the given acl, with all occurances of grantToFilter removed
filterAcl: function(acl, grantToFilter)
```